lib/errorUtils.js

```diff
@@ -0,0 +1,63 @@
+'use strict'
+
+const Assert = require('assert')
+
+const types = {
+  system: [
+
+      // JavaScript
+
+    EvalError,
+    RangeError,
+    ReferenceError,
+    SyntaxError,
+    TypeError,
+    URIError,
+
+      // Node
+
+    Assert.AssertionError
+  ]
+}
+
+function reflect (err, types) {
+  if (match(err, types)) {
+    toss(err)
+  }
+}
+
+function toss (err, rethrow) {
+  if (rethrow || canHandlePromiseRejection()) {
+    throw err
+  } else {
+    // fallback cause DEP0018
+    process.exit(1)
+  }
+}
+
+function isSystem (err) {
+  return match(err, types.system)
+}
```

Comment:
I think we don’t need the match function, as there is only one thing in types

Reply:
done